src/App.tsx

```diff
@@ -1,223 +1,196 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
 import { AuthContext } from './components/Auth/AuthContext';
+import {
+  ErrorNotification,
+} from './components/ErrorNotification/ErrorNotification';
+import { TodosSelection } from './components/TodosSelection/TodosSelection';
+import { TodoForm } from './components/TodoForm/TodoForm';
+import { TodoList } from './components/TodoList/TodoList';
+
+import { Todo } from './types/Todo';
+import { TodosFilter } from './types/TodosFilter';
+import { addTodo, getTodos, removeTodo } from './api/todos';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filtredTodos, setFiltredTodos] = useState<Todo[]>([]);
+  const [
+    statusToFilter,
+    setStatusToFilter,
+  ] = useState<TodosFilter>(TodosFilter.All);
+  const [isAdding, setIsAdding] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo>({
+    id: 0,
+    userId: 0,
+    title: '',
+    completed: false,
+  });
+  const [todoIdsToRemove, setTodoIdsToRemove] = useState<number[]>([]);
+
+  const closeNotification = useCallback(() => setHasError(false), []);
+
+  const filterTodosBy = useCallback((status: TodosFilter) => {
+    const filtredByStatusTodos = todos.filter(({ completed }) => {
+      switch (status) {
+        case TodosFilter.Active:
+          return !completed;
+
+        case TodosFilter.Completed:
+          return completed;
+
+        default:
+          return true;
+      }
+    });
+
+    setStatusToFilter(status);
+    setFiltredTodos(filtredByStatusTodos);
+  }, [todos]);
```

Comment:
Let's save filtered todos to constant instead of using this function, use useMemo hook to avoid unnecessary filtering